generate_network.py

```diff
@@ -105,79 +95,110 @@ def generate_scalefree_graph(n):
+
+
+def generate_star_graph(n):
+    return nx.star_graph(n)
 
 
 # Generate the network from nw type
-def generate_network(num_nodes, nw_type, prefix):
+def generate_network(num_nodes, nw_type):
     G = nx.empty_graph()
-    if nw_type == "configuration_model":
+    if nw_type == nw_types.configmodel:
         G = generate_config_model(num_nodes)
-    elif nw_type == "scalefree":
+    elif nw_type == nw_types.scalefree:
         G = generate_scalefree_graph(num_nodes)
-    elif nw_type == "newman_watts_strogatz":
+    elif nw_type == nw_types.newmanwattsstrogatz:
         G = generate_newman_watts_strogatz_graph(num_nodes) 
+    elif nw_type == nw_types.barbell:
+        G = generate_barbell_graph(num_nodes) 
+    elif nw_type == nw_types.balancedtree:
+        G = generate_balanced_tree(num_nodes) 
+    elif nw_type == nw_types.star:
+        G = generate_star_graph(num_nodes) 
     else: 
         print(nw_type +": Unsupported network type")
         sys.exit(1)
-    H = postprocess_network(G, prefix)
-    return H
+    return postprocess_network(G)
 
 
-# used by generate_dump_data, *ought* to be global to handle partitions
-ports_shifted = 0
-def postprocess_network(G, prefix):
-    G = nx.Graph(G)         # prune out parallel/multi edges
-    G.remove_edges_from(nx.selfloop_edges(G))   # Removing self-loops
-    # Labeling nodes to match waku containers
+# Label the generated network with prefix
+def postprocess_network(G):
+    G = nx.Graph(G)                             # prune out parallel/multi edges
+    G.remove_edges_from(nx.selfloop_edges(G))   # remove the self-loops
     mapping = {i: f"{prefix}{i}" for i in range(len(G))}
-    return nx.relabel_nodes(G, mapping)
+    return nx.relabel_nodes(G, mapping)         # label the nodes
+
 
+### file format related fns ###########################################################
+
+#Generate per node toml configs
+def generate_toml(topics, node_type=node_types.desktop):
+    topic_str =  " ". join(get_random_sublist(topics))    # topics as a space separated string
+    if node_type == node_type.desktop:
+        toml = "rpc-admin = true\nkeep-alive = true\n"
+    elif node_type == node_type.mobile:
+        toml =  "rpc-admin = true\nkeep-alive = true\n"
+    else:
+        print(node_type +": Unsupported node type")
+        sys.exit(1)
+    toml += f"topics = \"{topic_str}\"\n"
+    return toml
 
-# Generate dump data from the network and topics
-def generate_dump_data(H, topics):
-    data_to_dump = {}
-    global ports_shifted
+
+# Generates network-wide json and per-node toml and writes them 
+def generate_and_write_files(dirname, num_topics, H):
+    if not os.path.exists(dirname):
+        os.mkdir(dirname)
+    elif not os.path.isfile(dirname) and os.listdir(dirname):
+        print(dirname +": exists and is not empty")
+        sys.exit(1)
+    elif os.path.isfile(dirname):
+        print(dirname +": exists and is not a directory")
+        sys.exit(1)
```

Comment:
I would probably take the validation outside of this function into its own stuff, in the most outside layer possible.
That way you detach the side-effects and it can be reusable.

Reply:
Reusable, yes. I was trying to be tad stingy and omit the explicit "exists but empty" check, but lifted it to outer block and made it a function of its own.  :)